### ISSUE TITLE
Direct plots and other objects using IPython protocol to notebook

### ISSUE THREAD

**Issue by remram44** [VizierDB/web-api#1]
TLDR `%matplotlib inline` for Vizier :wink:

The "IPython protocol" is basically objects exposing `_repr_html_()`. IPython has a `display()` method that sends those to the notebook, and similarly does the same for the cell's "output" (last line).

To support everything that works in IPython:
* [ ] Use `_repr_html_()` to format rich objects for the notebook
* [ ] Need to make sure that calling `IPython.display.display()` sends objects to our frontend
  * `ipykernel` [sets a `DisplayPublisher` (and `InteractiveShell`)](https://github.com/ipython/ipykernel/blob/ebc13d81b0096684aff07ff60fa9d561a06f2d62/ipykernel/zmqshell.py#L60-L201). By their own admittance, this part of IPython is not very well factored.
* [ ] Take the output from the last line of a Python cell and similarly run it through `display()`
* Check how matplotlib works (do they just call `display()`?)
  * `ipykernel` sets `$MPLBACKEND` to [`module://ipykernel.pylab.backend_inline`](https://github.com/ipython/ipykernel/blob/ebc13d81b0096684aff07ff60fa9d561a06f2d62/ipykernel/pylab/backend_inline.py) with a `show()` that calls `display()`, so every figure created is sent to the notebook
  * Apache Zeppelin does [the same thing](https://github.com/apache/zeppelin/blob/1972a58620b16bc4ec54191fb8660cf274647bd8/interpreter/lib/python/backend_zinline.py)
  * [example](https://gist.github.com/remram44/c856324e3d285b52b074d60ad0ddb4e5)

**Comment by okennedy** [comment]
@mrb24 Implemented something analogous to this already.  Modulo documentation, perhaps this issue can be considered closed?
